stingray/lightcurve.py

```diff
@@ -175,3 +209,165 @@ def rebin_lightcurve(self, dt_new, method='sum'):
 
         lc_new = Lightcurve(bin_time, bin_counts)
         return lc_new
+
+    def join(self, other):
+        """
+        Join two lightcurves into a single object.
+
+        The new Lightcurve object will contain time stamps from both the
+        objects. The individual count per bin in the resulting object will
+        be the sum total of the count per bin in both lightcurve objects.
+
+        Parameters
+        ----------
+        other : Lightcurve object
+            The other Lightcurve object which is supposed to be joined with.
+
+        Returns
+        -------
+        lc_new : Lightcurve object
+            The resulting lightcurve object.
+
+        Example
+        -------
+        >>> time1 = [5, 10, 15]
+        >>> count1 = [300, 100, 400]
+        >>> time2 = [15, 20, 25]
+        >>> count2 = [600, 1200, 800]
+        >>> lc1 = Lightcurve(time1, count1)
+        >>> lc2 = Lightcurve(time2, count2)
+        >>> lc = lc1.join(lc2)
+        >>> lc.time
+        array([ 5, 10, 15, 20, 25])
+        >>> lc.counts
+        array([ 300,  100, 1000, 1200,  800])
+        """
+        assert self.dt == other.dt, "The bin width of both the lightcurves" \
+                                    " must be same."
+        # TODO : If different, rebin both the lightcurves with
+        # dt_new = LCM(self.dt, other.dt) and raise a warning.
```

Comment:
I don't see a real application of this behavior. I believe this case should raise an `Error` and not join, or raise a `Warning` and drop one of the duplicated time bin, in this case I would choose to keep the duplicate of lc1 (where the method is called) and drop the duplicate of lc2, though physically it also doesn't make sense to me so I would choose `Error`.

On the other hand this is the kind of operation that I would expect when summing light curves that do not completely overlap, but I would still expect a `Warning` from `SIMON`